packages/dom-adapters/src/utils/unwrapByToolType.ts

```diff
@@ -0,0 +1,116 @@
+import { getAbsoluteRangeOffset } from './getAbsoluteRangeOffset.js';
+import { splitElement } from './splitElement.js';
+
+/**
+ * Unwraps elements of the specified tool type from the passed range
+ *
+ * In order to work correctly, function requires that there are no nested elements of the same tool type
+ *
+ * There are three cases we need to consider:
+ * 1. Range is contained in the element of the specified tool type
+ * 2. Range contains element or elements of the specified tool type
+ * 3. Range intersects with the element of the specified tool type
+ *
+ * The second and the third cases could appear at the same time
+ *
+ * @param range - range to unwrap
+ * @param targetTool - tool to unwrap
+ */
+export function unwrapByToolType(range: Range, targetTool: string): void {
+  const commonAncestorElement = range.commonAncestorContainer instanceof Text ? range.commonAncestorContainer.parentElement! : range.commonAncestorContainer as HTMLElement;
+
+  /**
+   * To cover the first case, we need to check if there are any closest elements with targetTool relative to common ancestor
+   */
+  const targetToolAbove = commonAncestorElement.closest(`[data-tool="${targetTool}"]`) as HTMLElement | null;
```

Comment:
Many utility functions relies on `data-tool` attribute, which is a prat of business logic. Maybe we can do something with that implicit dependency